File: dotnet/src/Chalk.Ir/IrVersion.cs
namespace Chalk.Ir;

/// <summary>
/// The plan IR version this build speaks. Twin of <c>chalk.ir.IrVersion</c> in the planner
/// (<c>docs/design/02-ir.md</c> §2); the two constants must always agree.
/// </summary>
public static class IrVersion
{
    /// <summary>The version this client stamps on <c>PlanRequest.client_ir_version</c>.</summary>
    public const uint Current = 1;

    /// <summary>The oldest IR version this client can read.</summary>
    public const uint Minimum = 1;
}

File: planner/src/main/java/chalk/ir/IrVersion.java
package chalk.ir;

/**
 * The plan IR version this build speaks. Twin of {@code Chalk.Ir.IrVersion} on the client
 * (docs/design/02-ir.md §2); the two constants must always agree.
 */
public final class IrVersion {
  /** The version the planner stamps on every {@code Plan.ir_version}. */
  public static final int CURRENT = 1;

  /** Oldest client IR version this planner will serve. */
  public static final int MINIMUM = 1;

  private IrVersion() {}
}
